### PR TITLE
LPS-170788 Filter manually objects rest applications in the API explorer

### PR DESCRIPTION
**Pull request by LuismiBarcos**
## Purpose of the PR
Filtering the Object REST Applications created dynamically in the Liferay API Explorer.

When we have more than one instances, the objects created can be visible (only the name) among the different Liferay instances. This PR solves that.

### Before the PR
Instance 1: 
image

Instance 2:
image

### After the PR
Instance 1: image
Instance 2: 
image

### CODE REVIEW

**Comment by LuismiBarcos** [comment]
ci:test:sf

**Comment by LuismiBarcos** [comment]
ci:test:relevant

**Comment by 4lejandrito** [comment]
Thanks @LuismiBarcos. Use this LPS https://issues.liferay.com/browse/LPS-170788 I just created.

File: modules/apps/headless/headless-discovery/headless-discovery-impl/src/main/java/com/liferay/headless/discovery/internal/jaxrs/application/HeadlessDiscoveryOpenAPIApplication.java [hunk 1]
```diff
@@ -124,15 +185,59 @@ private void _addPaths(
 		}
 	}
 
+	private List<ApplicationDTO> _filterObjectsApplicationByCompany(
+		ApplicationDTO[] applicationDTOS) {
+
+		Stream<ApplicationDTO> applicationDTOStream = Arrays.stream(
```
Comment (ccorreagg) [inline modules/apps/headless/headless-discovery/headless-discovery-impl/src/main/java/com/liferay/headless/discovery/internal/jaxrs/application/HeadlessDiscoveryOpenAPIApplication.java:191]: I am not sure if Brian will like the Stream...

What do you think about using the TransformUtil class?

```
private List<ApplicationDTO> _filterObjectsApplicationByCompany(
	ApplicationDTO[] applicationDTOS) {

	return TransformUtil.transformToList(
		applicationDTOS,
		applicationDTO -> {
			if (_objectApplicationCompanyIdsMap.containsKey(
					applicationDTO.base)) {

				List<String> companyIds =
					_objectApplicationCompanyIdsMap.get(
						applicationDTO.base);

				if (companyIds.contains(
						String.valueOf(
							CompanyThreadLocal.getCompanyId()))) {

					return applicationDTO;
				}

				return null;
			}

			return applicationDTO;
		});
}
```

**Comment by LuismiBarcos** [comment]
I've changed the LPS to add the correct LPS number

**Comment by LuismiBarcos** [comment]
ci:forward

**Comment by LuismiBarcos** [comment]
Sent here https://github.com/brianchandotcom/liferay-portal/pull/127454

**Pull request closed without merging**